tools/cldr-unittest/src/org/unicode/cldr/unittest/TestCheckCLDR.java

```diff
@@ -765,88 +768,98 @@ public String getXpath() {
             return;
         }
-        
-        CLDRConfig conf = testInfo;
-        Factory factory = conf.getMainAndAnnotationsFactory();
-        File[] paths = {
-            new File(CLDRPaths.LAST_COMMON_DIRECTORY + "main/"),
-            new File(CLDRPaths.LAST_COMMON_DIRECTORY + "annotations/"),
-        };
-        Factory factoryOld = SimpleFactory.make(paths, ".*");
 
-        CLDRFile english = factory.make("en", true);
-        CLDRFile oldEnglish = factoryOld.make("en", true);
-        Set<String> newPaths = new TreeSet<>();
-        Set<String> diffValues = new TreeSet<>();
-        boolean firstFail = true;
-        
-        UnicodeSet okAnnotations = new UnicodeSet("[🤖 😺 😸 😹 😻-😽 🙀 😿 😾 🐺 🦊 🦁 🦄 🐹 🐻 🐼 🐸 🧩 ⭕]").freeze();
-        
-        for (String path : english) {
-            if (path.contains("/alias")) {
-                continue;
-            }
-            String value = english.getStringValue(path);
-            String valueOld = oldEnglish.getStringValue(path);
-            boolean isDiff = false;
-            if (valueOld == null) {
-                newPaths.add(path);
-            } else if (value != null 
-                && !value.equalsIgnoreCase(valueOld)
-                && !path.startsWith("//ldml/localeDisplayNames/types/type[@key=\"m0\"]")) {
-                diffValues.add(path);
-                isDiff = true;
-            }
-            boolean regexMatch = SubmissionLocales.pathAllowedInLimitedSubmission(path);
-            if (isDiff && !regexMatch) {
-                errln((firstFail ? "Add these to regex in SubmissionLocales OR filter from this test if unimportant:\n\t\t\t" : "")
-                    + "Match fails with " + path + ", old: " + valueOld + ", new: " + value);
-                firstFail = false;
-            } else if (!isDiff && regexMatch) {
-                
-                // will need to change this for new releases!!
-
-                if (path.endsWith("/listPatternPart[@type=\"start\"]") 
-                    || path.endsWith("/listPatternPart[@type=\"middle\"]") 
-                    || path.endsWith("[@type=\"tts\"]") 
-                    || okAnnotations.containsSome(path)
-                    ) {
-                    continue;
-                }
-                errln("Match overshoots with " + path 
-                    + (firstFail ? "\nAdd these to regex in SubmissionLocales or filter from this test if unimportant" : ""));
+        assertTrue(language + " locale is " + (inclusionDesired ? "" : "not ") + "in cldrLocales", cldrLocales.contains(locale) == inclusionDesired);
+
+        // check for expected script
+        String max = likely.maximize(locale);
+        assertNotNull(language + " locale has script", max);
+        String maxScript = max == null ? null : ltp.set(max).getScript();
+
+        String desiredScript = "Latn";
+        switch(language) {
+        case "mai": case "Deva": case "kok": desiredScript = "Deva"; break;
```

Comment:
Can "Deva" be a language as well as a script?

Reply:
A number of the changes in the tests were to help address some of the points that you brought up, so thanks for that.

As to Deva, that's a good catch. A typo on my part since I originally was testing locales, and using sd_Deva. It's harmless, since the code space of scripts and languages are disjoint, but I'll fix the next time I touch the file.